fl4health/preprocessing/warmed_up_module.py

```diff
@@ -0,0 +1,126 @@
+from typing import Optional
+
+import torch
+from flwr.common.logger import log
+
+
+class WarmedUpModule:
+    """This class is used to load a pretrained model into the target model."""
+
+    def __init__(
+        self,
+        pretrained_model: Optional[torch.nn.Module] = None,
+        pretrained_model_path: Optional[Path] = None,
+        weights_mapping_path: Optional[Path] = None,
+    ) -> None:
+        """Initialize the WarmedUpModule with the pretrained model states and weights mapping dict.
+
+        Args:
+            pretrained_model (Optional[torch.nn.Module]): Pretrained model.
+                                                          This is mutually exclusive with pretrained_model_path.
+            pretrained_model_path (Optional[Path]): Path of the pretrained model.
+                                                    This is mutually exclusive with pretrained_model.
+            weights_mapping_dir (Optional[str], optional): Path of to json file of the weights mapping dict.
+            If models are not exactly the same, a weights mapping dict is needed to map the weights of the pretrained
+            model to the target model.
+        """
+        if pretrained_model is not None and pretrained_model_path is not None:
+            AssertionError(
+                "pretrained_model_path and pretrained_model is mutually exclusive. Please provide one of them."
+            )
+
+        elif pretrained_model is not None:
+            log(INFO, "Pretrained model is provided.")
+            self.pretrained_model_state = pretrained_model.state_dict()
+
+        elif pretrained_model_path is not None:
+            assert os.path.exists(
+                pretrained_model_path
+            ), f"Pretrained model path {pretrained_model_path} does not exist."
+            log(INFO, f"Loading pretrained model from {pretrained_model_path}")
+            self.pretrained_model_state = torch.load(pretrained_model_path).state_dict()
+
+        else:
+            raise AssertionError("At least one of pretrained_model_path and pretrained_model should be provided.")
+
+        if weights_mapping_path is not None:
+            with open(weights_mapping_path, "r") as file:
+                self.weights_mapping_dict = json.load(file)
+        else:
+            log(INFO, "Weights mapping dict is not provided. Matching states directly, based on target model's keys.")
+            self.weights_mapping_dict = None
+
+    def get_matching_component(self, key: str) -> Optional[str]:
+        """Get the matching component of the key from the weights mapping dictionary. Since the provided mapping
+        can contain partial names of the keys, this function is used to split the key of the target model and
+        match it with the partial key in the mapping, returning the complete name of the key in the pretrained model.
+
+        This allows users to provide one mapping for multiple statistics that share the same prefix. For example,
```

Comment:
Do you mean "model states" instead of statistics?

Reply:
Yeah I think states make more sense.